packages/sources/stader-address-list/src/endpoint/address.ts

```diff
@@ -51,50 +88,314 @@ export const inputParameters: InputParameters<TInputParameters> = {
+    blockTag,
     network,
     chainId,
-    confirmations,
-  )
-  const filteredAddressList = addressList.filter(
-    ({ registrationStatus }) => registrationStatus === includedRegistrationStatus,
-  )
-  const result = {
-    data: {
-      validatorStatus,
-      result: filteredAddressList,
-    },
-    statusCode: 200,
+    batchSize,
+  }: {
+    poolFactoryManager: ethers.Contract
+    blockTag: number
+    network: string
+    chainId: string
+    batchSize: number
+  }): Promise<ValidatorAddress[]> {
+    try {
+      logger.debug('Fetching validator address list')
+      const poolCount = await poolFactoryManager.poolCount({ blockTag })
+      logger.debug(`Pool Count: ${poolCount}`)
+      const addressList: ValidatorAddress[] = []
+      for (let i = 1; i <= poolCount; i++) {
+        const nodeRegistryAddress: string = await poolFactoryManager.getNodeRegistry(i, {
+          blockTag,
+        })
+        const nodeRegistryManager = new ethers.Contract(
+          nodeRegistryAddress,
+          StaderNodeRegistryContract_ABI,
+          this.provider,
+        )
+        const validatorCount = (await nodeRegistryManager.nextValidatorId({ blockTag })) - 1
+        logger.debug(
+          `${validatorCount} addresses in pool ${i}. May not be the number that are active.`,
+        )
+        // Calculate number of pages based on total validators and batch size
+        const pages = calculatePages(validatorCount, batchSize)
+        for (let j = 1; j <= pages; j++) {
+          const validators = (await nodeRegistryManager.getAllActiveValidators(j, batchSize, {
+            blockTag,
+          })) as validatorsRegistryResponse[]
+          validators.forEach(([status, pubkey, , , withdrawVaultAddress, operatorId, , ,]) => {
+            addressList.push({
+              address: pubkey,
+              withdrawVaultAddress,
+              network,
+              chainId,
+              operatorId: Number(operatorId),
+              poolId: i,
+              status,
+            })
+          })
+        }
+      }
+      return addressList
+    } catch (e) {
+      logger.error({ error: e })
+      throw Error('Failed to retrieve validator addresses from contract')
+    }
   }
 
-  return Requester.success(jobRunID, result, true)
+  // Fetch socializing pool addresses
+  async fetchSocializingPoolAddresses({
+    poolFactoryManager,
+    blockTag,
+  }: {
+    poolFactoryManager: ethers.Contract
+    blockTag: number
+  }): Promise<PoolAddress[]> {
+    try {
+      logger.debug('Fetching socializing pool address list')
+      const poolCount = await poolFactoryManager.poolCount({ blockTag })
+      let socialPoolAddresses: PoolAddress[] = []
+      for (let i = 1; i <= poolCount; i++) {
+        const address = await poolFactoryManager.getSocializingPoolAddress(i, { blockTag })
+        socialPoolAddresses.push({ address, poolId: i })
+      }
+      socialPoolAddresses = filterDuplicates<PoolAddress>(socialPoolAddresses)
```

Comment:
I think this is something we could abstract as well
```typescript
const runAllSequentially = <T>(count, fn: (i: number) => Promise<T>): Promise<T[]> => {
	const results: T[] = []
	for (let i = 1; i <= count; i++) {
		results.push(await fn(i))
	}
	return results
}
```
And then
```suggestion
      const rawSocialPoolAddresses = await runAllSequentially(poolCount, i => poolFactoryManager.getSocializingPoolAddress)
      const socialPoolAddresses = filterDuplicates(socialPoolAddresses) 
      // Note the lack of explicit generic above, should be inferred by the param
```